src/carousel.tsx

```diff
@@ -0,0 +1,727 @@
+                transform: 'scale(1.5)', // 6px
+            },
+        },
+    },
+}));
+
+type PageBulletsProps = {
+    currentIndex: number;
+    numPages: number;
+    onPress?: (index: number) => void;
+};
+
+export const PageBullets: React.FC<PageBulletsProps> = ({currentIndex, numPages, onPress}) => {
+    const isInverse = useIsInverseVariant();
+    const classes = useBulletsStyles({isInverse});
+    const {isDesktopOrBigger} = useScreenSize();
+    return (
+        <Inline space={isDesktopOrBigger ? 16 : 8} alignItems="center">
+            {Array.from({length: numPages}, (_, i: number) => (
+                <Touchable
+                    key={i}
+                    maybe
+                    onPress={isDesktopOrBigger && onPress ? () => onPress(i) : undefined}
+                >
+                    <div className={classNames(classes.bullet, {active: i === currentIndex})} />
+                </Touchable>
+            ))}
+        </Inline>
+    );
+};
+
+const hideScrollbar = {
+    scrollbarWidth: 'none', // Hide in FF
+    '&::-webkit-scrollbar': {
+        display: 'none', // Hide in Chrome/Safari
+    },
+};
+
+const arrowButtonSize = 40;
+const arrowButtonStyle = (theme: Theme) => ({
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    width: arrowButtonSize,
+    height: arrowButtonSize,
+    borderRadius: '50%',
+    backgroundColor: theme.colors.backgroundContainer,
+    border: `1px solid ${theme.colors.border}`,
+    transition: 'opacity 0.2s',
+    '&[disabled]': {
+        opacity: 0,
+    },
+    // don't show carrousel arrow buttons in touch devices, just regular horizontal scroll
+    '@media (pointer: coarse)': {
+        display: 'none',
+    },
+});
+
+const useStyles = createUseStyles((theme) => ({
+    carouselContainer: {
+        position: 'relative',
+    },
+    arrowButton: {
+        ...arrowButtonStyle(theme),
+        position: 'absolute',
+        zIndex: 1,
+        top: `calc(50% - ${arrowButtonSize / 2}px)`,
+        '&.prev': {
+            left: -arrowButtonSize / 2,
+            [theme.mq.tabletOrSmaller]: {
+                left: ({sideMargin}) => -sideMargin,
+            },
+            [theme.mq.largeDesktop]: {
+                left: -(24 + arrowButtonSize),
+            },
+        },
+        '&.next': {
+            right: -arrowButtonSize / 2,
+            [theme.mq.tabletOrSmaller]: {
+                right: ({sideMargin}) => -sideMargin,
+            },
+            [theme.mq.largeDesktop]: {
+                right: -(24 + arrowButtonSize),
+            },
+        },
+    },
+    carousel: {
+        display: 'flex',
+        overflowX: 'auto',
+        minWidth: '100%',
+        scrollSnapType: ({free}) => (free ? 'initial' : 'x mandatory'),
+        ...hideScrollbar,
+        [theme.mq.tabletOrSmaller]: {
+            margin: ({sideMargin}) => `0 -${sideMargin}px`,
+            '&::before, &::after': {
+                content: '""',
```

Comment:
what does this content hack do? (or what effect does in this place?)

Reply:
The `::before` and `::after` pseudoelements are used here to generate the extra spacing to center the carousel items in the `CenteredCarousel`
![image](https://user-images.githubusercontent.com/1849576/153558935-9e055908-c6a0-47d8-af86-f037903eb349.png)